lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -535,49 +647,84 @@ namespace Js
-            default:
-                AssertMsg(false, "We should never be here");
-                return library->GetUndefined();
+        case TypeIds_ArrayBuffer:
+            return library->CreateStringFromCppLiteral(_u("[object ArrayBuffer]"));
+        case TypeIds_Int8Array:
+            return library->CreateStringFromCppLiteral(_u("[object Int8Array]"));
+        case TypeIds_Uint8Array:
+            return library->CreateStringFromCppLiteral(_u("[object Uint8Array]"));
+        case TypeIds_Uint8ClampedArray:
+            return library->CreateStringFromCppLiteral(_u("[object Uint8ClampedArray]"));
+        case TypeIds_Int16Array:
+            return library->CreateStringFromCppLiteral(_u("[object Int16Array]"));
+        case TypeIds_Uint16Array:
+            return library->CreateStringFromCppLiteral(_u("[object Uint16Array]"));
+        case TypeIds_Int32Array:
+            return library->CreateStringFromCppLiteral(_u("[object Int32Array]"));
+        case TypeIds_Uint32Array:
+            return library->CreateStringFromCppLiteral(_u("[object Uint32Array]"));
+        case TypeIds_Float32Array:
+            return library->CreateStringFromCppLiteral(_u("[object Float32Array]"));
+        case TypeIds_Float64Array:
+            return library->CreateStringFromCppLiteral(_u("[object Float64Array]"));
+        case TypeIds_Symbol:
+        case TypeIds_SymbolObject:
+            return library->CreateStringFromCppLiteral(_u("[object Symbol]"));
+        case TypeIds_Map:
+            return library->CreateStringFromCppLiteral(_u("[object Map]"));
+        case TypeIds_Set:
+            return library->CreateStringFromCppLiteral(_u("[object Set]"));
+        case TypeIds_WeakMap:
+            return library->CreateStringFromCppLiteral(_u("[object WeakMap]"));
+        case TypeIds_WeakSet:
+            return library->CreateStringFromCppLiteral(_u("[object WeakSet]"));
+        case TypeIds_Generator:
+            return library->CreateStringFromCppLiteral(_u("[object Generator]"));
+        default:
+            AssertMsg(false, "We should never be here");
+            return library->GetUndefined();
         }
     }
 
     Var JavascriptObject::ToStringHelper(Var thisArg, ScriptContext* scriptContext)
     {
         TypeId type = JavascriptOperators::GetTypeId(thisArg);
         JavascriptLibrary* library = scriptContext->GetLibrary();
+
+        if (type == TypeIds_HostDispatch)
+        {
+            RecyclableObject* hostDispatchObject = RecyclableObject::FromVar(thisArg);
+            DynamicObject* remoteObject = hostDispatchObject->GetRemoteObject();
+            if (!remoteObject)
+            {
+                Var result = nullptr;
+                Js::Var values[1];
+                Js::CallInfo info(Js::CallFlags_Value, 1);
+                Js::Arguments args(info, values);
+                values[0] = thisArg;
+                if (hostDispatchObject->InvokeBuiltInOperationRemotely(EntryToString, args, &result))
```

Comment:
we should be able to use the LoadScriptFile(, "crossthread") to test test. Did we add EntryToString operation in the remote support in es5 timeframe?